invokeai/frontend/web/src/features/gallery/components/ImageMetadataViewer/ImageMetadataActions.tsx

```diff
@@ -180,6 +195,19 @@ const ImageMetadataActions = (props: Props) => {
             );
           }
         })}
+      {metadata.controlnets &&
+        metadata.controlnets.map((controlnet, index) => {
+          if (isValidControlNetModel(controlnet.control_model)) {
```

Comment:
Let's create the list of valid controlnets outside the render function, inside a `useMemo()`. Then, in the render function, we can map over the valid ones without filtering.